src/mqt/bench/error_correction/steane_transpiler.py

```
@@ -0,0 +1,357 @@
+    get_seven_qubit_steane_code_encoding_circuit,
+    get_seven_qubit_steane_code_syndrome_extraction_circuit,
+)
+
+if TYPE_CHECKING:
+    from qiskit.circuit import CircuitInstruction
+
+
+class SteaneTranspiler:
+    """A high-level transpiler that encodes a QuantumCircuit using Steane's 7-qubit error correction code."""
+
+    def __init__(self, original_circuit: QuantumCircuit, add_syndromes: bool = True) -> None:
```

Ruff (0.15.15)

[warning] Boolean-typed positional argument in function definition

(FBT001)